fe/fe-core/src/main/java/com/starrocks/connector/iceberg/io/IcebergCachingFileIO.java

```diff
@@ -155,6 +162,60 @@ public SeekableInputStream toSeekableInputStream() {
         public boolean isExistOnDisk() {
             return this.inputFile != null && this.inputFile.exists();
         }
+
+        public void pin() {
+            useCount += 1;
+        }
+        public void unpin() {
+            useCount -= 1;
+        }
+
+        public static DiskCacheEntry newDiskCacheEntry(SeekableInputStream stream, long fileLength, String key) {
+            long totalBytesToRead = fileLength;
+            HadoopOutputFile tmpOutputFile = (HadoopOutputFile) IOUtil.getTmpOutputFile(
+                    METADATA_CACHE_DISK_PATH, key);
+            PositionOutputStream outputStream = tmpOutputFile.createOrOverwrite();
+            Path localFilePath = new Path(IOUtil.remoteToLocalFilePath(METADATA_CACHE_DISK_PATH, key));
+            try {
+                while (totalBytesToRead > 0) {
+                    // read the stream in 4MB chunk
+                    int bytesToRead = (int) Math.min(BUFFER_CHUNK_SIZE, totalBytesToRead);
```

Comment:
I think you could read all length when write disk cache.

Reply:
this interface is also can be used when we need fetch file from remote to disk directly, in that case read by chunk is better. and I think it's also ok (not bad) when we need writer data from memory to disk.